xtask/src/ast_src.rs

```diff
@@ -607,55 +1972,165 @@ pub(crate) const AST_SRC: AstSrc = AstSrc {
             const_args: [ConstArg],
             T![>]
         }
+
+        /// Type argument that is passed at generic instantiation site.
+        ///
+        /// ```
+        /// use foo::<'a, ❰ u64 ❱, ❰ bool ❱, Item = Bar, 42>::baz;
+        /// ```
+        ///
+        /// [Reference](https://doc.rust-lang.org/reference/paths.html#paths-in-expressions)
         struct TypeArg { TypeRef }
+
+        // TODO: verify inline type bounds example
+        /// Associated type argument that is passed at generic instantiation site.
+        /// ```
+        /// use foo::<'a, u64, bool, ❰ Item = Bar ❱, 42>::baz;
+        ///
+        /// trait Bruh<T>: Iterator<❰ Item: Debug ❱> {}
+        /// ```
+        ///
         struct AssocTypeArg : TypeBoundsOwner { NameRef, T![=], TypeRef }
+
+        // TODO: verify?
+        /// Lifetime argument that is passed at generic instantiation site.
+        ///
+        /// ```
+        /// fn foo<'a>(s: &'a str) {
+        ///     bar::<❰ 'a ❱>(s);
+        /// }
+        /// ```
+        ///
+        /// [Reference](https://doc.rust-lang.org/reference/paths.html#paths-in-expressions)
         struct LifetimeArg { T![lifetime] }
+
+        // TODO: does this peratain to const generics?
+        // What does equal sign do here?
+        /// Constant value argument that is passed at generic instantiation site.
+        ///
+        /// ```
+        /// foo::<❰ u32 ❱, ❰ true ❱ >();
+        ///
+        /// bar::<❰ { 2 + 2} ❱>();
+        /// ```
+        ///
+        /// [RFC](https://github.com/rust-lang/rfcs/blob/master/text/2000-const-generics.md#declaring-a-const-parameter)
         struct ConstArg { Literal, T![=], BlockExpr }
 
-        struct MacroItems: ModuleItemOwner{ }
+        // TODO: Idk what I am writing here, please don't believe these words.
+        // TODO: clarify @matklad @edwin0cheng
```

Comment:
@matklad Maybe we could use ItemList instead inside mbe ?

Reply:
Maybe... The `ItemList` also contains the `{}`, but maybe that's not important?

Reply:
So is my doc comment correct or not? I would like to avoid doing such refactorings in this PR, let's agree with this doc or change it for the better.
cc @edwin0cheng @matklad

Reply:
Let leave a FIXME here to remove it for future. Something like:
```
FIXME: (@edwin0cheng) Remove it to use ItemList instead 
```

Reply:
Thanks, converted this to FIXME in f5e2e02aa9864f5c929ff36f99cde5fd4be2544e